src/plugins/shared-variables/slate/variables-plugin.tsx

```diff
@@ -1,284 +1,121 @@
-import { Variable, VariableChip, VariableType } from "@concord-consortium/diagram-view";
-import { kVariableSlateType, getVariables, getOrFindSharedModel } from "./variables-text-content";
+import { VariableChip, VariableType } from "@concord-consortium/diagram-view";
+import { getVariables } from "./variables-text-content";
 import { TextContentModelType } from "../../../models/tiles/text/text-content";
+import { TextContentModelContext } from "../../../models/tiles/text/text-content-context";
 
 const kVariableClass = "slate-variable-chip";
+const kSlateVoidClass = "cc-slate-void";
+const kVariableFormat = "clueVariable";
 
-function parseVariableValue(value?: string) {
-  return value ? parseFloat(value) : undefined;
+// FIXME: Clean up these types and this interface.
+export function VariablesPlugin(textContent: TextContentModelType): any {
+  return {
+    onInitEditor: (editor: CustomEditor) => withClueVariables(editor, textContent) 
+  };
 }
 
-// This is for the input field
-function variableValueToString(value?: number) {
-  if (value === undefined) {
-    return "";
-  }
-  // The first argument is the locale, using undefined means it should pick up the default
-  // browser locale
-  return new Intl.NumberFormat(undefined, { maximumFractionDigits: 4 }).format(value);
+export interface VariableElement extends BaseElement { 
+  type: "clueVariable",
+  reference: string,
+  children: EditorValue
 }
 
-interface IRenderOptions {
-  textTile: TextContentModelType;
-  isSerializing?: boolean;
-  isHighlighted?: boolean;
-  onClick?: () => void;
-  onDoubleClick?: () => void;
-}
-function renderVariable(node: Inline, attributes: RenderAttributes, children: ReactNode, options?: IRenderOptions) {
-  const { data } = node;
-  const { className, ...otherAttributes } = attributes;
-  const { isHighlighted, isSerializing, onClick: _onClick, onDoubleClick: _onDoubleClick } = options || {};
-  const selectedClass = isHighlighted && !isSerializing ? "slate-selected" : undefined;
-  const reference: string = data.get("reference");
-  const classes = classNames(classArray(className), kSlateVoidClass, kVariableClass) || undefined;
-  const onClick = isSerializing ? undefined : _onClick;
-  const onDoubleClick = isSerializing ? undefined : _onDoubleClick;
+export const isVariableElement = (element: BaseElement): element is VariableElement => {
+  return element.type === kVariableFormat;
+};
 
-  if (!options) {
-    throw new Error("Can't render variable without options");
+export const insertTextVariable = (variable: VariableType, editor?: Editor) => {
+  if (!editor) {
+    console.warn("inserting variable but there is no editor");
+    return;
   }
+  const reference = variable.id;
+  const varElt: VariableElement = { type: "clueVariable", reference, children: [{text: "" }]};
+  Transforms.insertNodes(editor, varElt);
+};
+
+export const findSelectedVariable = (selectedElements: any, variables: VariableType[]) => {
+  let selected = undefined;
```

Comment:
The return type of selectedElements in slate needs updating. It returns a NodeEntry list which takes the form [Node, Path] not a BaseElement list. NodeEntry[] is a possibility. 

NodeEntry is pretty slate specific so it is possible we'd want to wrap it up in a type we define in the slate editor instead.  This also makes me wonder if we want to expose any location based info in slate (in this case the Path). Here we don't need it.

Another option would be to add a util to slate that say "find me all elements of a type" in the selection, line, or whole text (or some defined set of regions). Then this iteration would disappear.  There are a couple other places I did something similar to this that are worth looking at to inform the decision. One is in `getTileTextVariables` in variables-text-content.ts where we're trying to find all variable elements in the whole editor/text area.